src/bentoml_io/models.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import inspect
+import sys
+import typing as t
+from typing import ClassVar
+
+from pydantic import BaseModel
+from pydantic import Field
+from pydantic import RootModel
+from pydantic import create_model
+from typing_extensions import get_args
+
+from .typing_utils import is_file_like
+from .typing_utils import is_image_type
+from .typing_utils import is_iterator_type
+from .typing_utils import is_list_type
+
+if t.TYPE_CHECKING:
+    from starlette.requests import Request
+    from starlette.responses import Response
+
+    from .serde import Serde
+
+
+class IOMixin:
+    multipart_fields: ClassVar[list[str]]
+
+    @classmethod
+    def __pydantic_init_subclass__(cls) -> None:
+        from pydantic._internal._typing_extra import is_annotated
+
+        cls.multipart_fields = []
+        for k, field in cls.model_fields.items():
+            annotation = field.annotation
+            if is_list_type(annotation):
+                annotation = get_args(annotation)[0]
+            if is_annotated(annotation):
+                annotation = get_args(annotation)[0]
+            if is_file_like(annotation) or is_image_type(annotation):
+                cls.multipart_fields.append(k)
+
+    @classmethod
+    async def from_http_request(cls, request: Request, serde: Serde) -> IODescriptor:
```

Comment:
```suggestion
    async def from_http(cls, request: Request, serde: Serde) -> IODescriptor:
```
Should we do `from_http` and `to_http`?

Reply:
`from_http` doesn't tell if it's from request or response